### PR TITLE
Batch transaction error handling

### PR DESCRIPTION
**Pull request by RunyaoFan**
Creates batch transaction rollback mechanism in case of sub-request failure

### CODE REVIEW

**Comment by edwin-huber** [comment]
Copying a partition to validate the changes against it is a nice idea, but I am concerned that this might generate excessive load when Azurite has been running for a while with some of our SDKs.  
We need to test this with Durable Functions and logic apps, which make heavy use of batch requests, as well as it's behavior with larger partitions.

**Comment by edwin-huber** [comment]
Currently working on some perf improvements for this PR as agreed with @XiaoningLiu

File: src/table/batch/TableBatchOrchestrator.ts [hunk 1]
```diff
@@ -75,34 +78,99 @@ export default class TableBatchOrchestrator {
...
       const request: BatchRequest = new BatchRequest(operation);
       this.requests.push(request);
     });
 
     let contentID = 1; // ToDO: validate contentID starts at 1 for batch
     if (this.requests.length > 0) {
+      // TO DO: the way used to get accountName, tableName, and partition key
+      // may not work if incoming request is not compliant with the right format
+      const reqUrl = this.requests[0].getUrl();
+      const accountName = reqUrl.split("/")[3];
```
Comment (XiaoningLiu) [inline src/table/batch/TableBatchOrchestrator.ts:92]: Azurite supports 2 kinds of URL pattern.
- Path style. Storage account name is in URL path. Like "http:/localhost:10000/accountname"
- Production style. Storage account name is in first segment of URL endpoint. Like "http:://account.suffix/

Comment (XiaoningLiu) [reply src/table/batch/TableBatchOrchestrator.ts:92]: Refer to https://github.com/azure/azurite/blob/1ab42a545be0eda8a9d2d10e967e24e327374278/src/table/middleware/tableStorageContext.middleware.ts#L74 for existing method to extract account name and table name.

File: src/table/generated/handlers/ITableHandler.ts [hunk 2]
```diff
@@ -17,13 +17,13 @@ export default interface ITableHandler {
...
-  queryEntities(table: string, options: Models.TableQueryEntitiesOptionalParams, context: Context): Promise<Models.TableQueryEntitiesResponse>;
-  queryEntitiesWithPartitionAndRowKey(table: string, partitionKey: string, rowKey: string, options: Models.TableQueryEntitiesWithPartitionAndRowKeyOptionalParams, context: Context): Promise<Models.TableQueryEntitiesWithPartitionAndRowKeyResponse>;
-  updateEntity(table: string, partitionKey: string, rowKey: string, options: Models.TableUpdateEntityOptionalParams, context: Context): Promise<Models.TableUpdateEntityResponse>;
-  mergeEntity(table: string, partitionKey: string, rowKey: string, options: Models.TableMergeEntityOptionalParams, context: Context): Promise<Models.TableMergeEntityResponse>;
-  deleteEntity(table: string, partitionKey: string, rowKey: string, ifMatch: string, options: Models.TableDeleteEntityOptionalParams, context: Context): Promise<Models.TableDeleteEntityResponse>;
+  queryEntities(table: string, options: Models.TableQueryEntitiesOptionalParams, context: Context, batchID?: string): Promise<Models.TableQueryEntitiesResponse>;
```
Comment (XiaoningLiu) [inline src/table/generated/handlers/ITableHandler.ts:20]: We should not manually change files under "generated folder". As we have done in another PR, batchId is included in TableContext.
